packages/connection-form/src/components/connection-form.tsx

```diff
@@ -626,96 +510,75 @@ function ConnectionForm({
-                    ...cloneDeep(initialConnectionInfo),
-                    connectionOptions: cloneDeep(connectionOptions),
-                  });
-                } else {
-                  setSaveConnectionModal('save');
-                }
-              }}
-              onSaveAndConnectClicked={() => {
-                setSaveConnectionModal('saveAndConnect');
-              }}
-              onConnectClicked={() => onSubmitForm('connect')}
-            />
-          )}
+                ))
+            }
+            onSaveAndConnect={onSubmitForm}
+          />
         </div>
       </form>
-
-      {showFavoriteActions && (
-        <SaveConnectionModal
-          open={saveConnectionModal !== 'hidden'}
-          saveText={
-            saveConnectionModal === 'saveAndConnect' ? 'Save & Connect' : 'Save'
-          }
-          onCancelClicked={() => {
-            setSaveConnectionModal('hidden');
-          }}
-          onSaveClicked={async (favoriteInfo: ConnectionFavoriteOptions) => {
-            setSaveConnectionModal('hidden');
-
-            const connectionInfo = getConnectionInfoToSave(favoriteInfo);
-            await callOnSaveConnectionClickedAndStoreErrors(connectionInfo);
-
-            if (saveConnectionModal === 'saveAndConnect') {
-              // Connect to the newly created favorite
-              onSubmitForm('connect');
-            }
-          }}
-          key={initialConnectionInfo.id}
-          initialFavoriteInfo={initialConnectionInfo.favorite}
-        />
-      )}
     </ConfirmationModalArea>
   );
 }
 
-const ConnectionFormWithPreferences = (
-  props: ConnectionFormPropsWithoutPreferences & {
-    preferences?: Partial<ConnectionFormPreferences>;
-  }
-) => {
-  const { preferences, ...rest } = props;
-
+const ConnectionFormWithSettings: React.FunctionComponent<
+  ConnectionFormPropsWithoutSettings & Partial<ConnectionFormSettings>
+> = ({
+  showFavoriteActions,
+  showHelpCardsInForm,
+  showPersonalisationForm,
+  protectConnectionStrings,
+  forceConnectionOptions,
+  showKerberosPasswordField,
+  showOIDCDeviceAuthFlow,
+  enableOidc,
+  enableDebugUseCsfleSchemaMap,
+  protectConnectionStringsForNewConnections,
+  showOIDCAuth,
+  showKerberosAuth,
+  showCSFLE,
+  showProxySettings,
+  saveAndConnectLabel,
+  ...rest
+}) => {
   return (
-    <ConnectionFormPreferencesContext.Provider value={preferences ?? {}}>
+    <ConnectionFormSettingsContext.Provider
+      value={{
+        showFavoriteActions,
+        showHelpCardsInForm,
+        showPersonalisationForm,
+        protectConnectionStrings,
+        forceConnectionOptions,
+        showKerberosPasswordField,
+        showOIDCDeviceAuthFlow,
+        enableOidc,
+        enableDebugUseCsfleSchemaMap,
+        protectConnectionStringsForNewConnections,
+        showOIDCAuth,
+        showKerberosAuth,
+        showCSFLE,
+        showProxySettings,
+        saveAndConnectLabel,
+      }}
```

Comment:
Before we could've passed this value straight to provider without any extra code because the outer code made sure that the object is stable and is not re-created every render, now that we're destructuring here, it would be better to make sure that the value is stable (even if it makes the code a bit weird), so let's wrap the value in useMemo before passing to provider, kinda like this:

```jsx
const value = useMemo(() => {
  return {
    ... properties
  }
}, [ ... dependencies])

return <Provider value={value}> ... </Provider>
```

Otherwise anything using the context will be re-rendering erroneously